draft-ietf-protected-audience-key-value-services.md

```diff
@@ -579,9 +613,51 @@ The output is a `response` to be sent to a Client.
    `rctxt`, as described in {{response-encryption}}. Return an empty `response` on failure.
 9.  Return `response`.
 
-### Parsing a Response
-
-TODO
+### Parsing a Response (#response-parsing)
+
+This section describes how a conforming Client MUST parse and validate a response from a Trusted Key Value service. 
+
+It takes as input the `request context` returned from {{request-generate}} in addition to the `encrypted response`.
+
+The output is a `result` map, where the keys are strings, and the values are maps with both keys and values as strings.
+
+1. Use `request context` as the context to decrypt `encrypted response` and obtain `framed response`, returning failure if decryption fails.
+1. Remove and extract the first 5 bytes from `framed response` as the framing header (described in {{framing}}), removing them from `framed response`.
+1. If `framing header`'s `Version` field is not 0 or 2, return failure.
+1. Let `length` be equal to the `framing header`'s `Size` field.
+1. If `length` is greater than the length of the remaining bytes in `framed response`, return failure.
+1. Take the first `length` remaining bytes in `framed response` as `serialized response`, discarding the rest.
+1. [CBOR] decode the `serialized response` into `response`, returning failure if decoding fails.
+1. If `response` is not a map, return failure.
+1. If `response["compressionGroups"]` does not exist, or is not an array, return failure.
+1. Let `results` be an empty array.
+1. For each `group` in `response["compressionGroups"]`:
+    1. If `group` is not a map, return failure.
+    1. If `group["content"]` does not exist, return failure.
+    1. If `framing header`'s `Version` field is 0:
```

Comment:
You could probably combine these cases to something like:

Set `serialized content` to the result of decompressing `group["content"]` according to the compression algorithm specified in the `framing header`'s `Compression` field, returning failure if decompression fails.

Reply:
Done.